board: arm64: Add suport for ROC-RK3588-PC

KouweiLee opened the pull request:
Add initial support for the ROC-RK3588-PC AArch64 board. 
It features a quad-core Cortex-A55 CPU based on the ARMv8.2 architecture. 
This commit also enables SMP, allowing all four cores to run the synchronization sample.

In future, we will add support for the AArch32-mode zephyr on RK3588-PC. Zephyr will run as an aarch32 guest vm on the aarch64 hypervisor.

KouweiLee committed:
dts: rockchip: add rk3588

Add initial device tree support for the Rockchip RK3588 SoC.
The DTS describes:
- Four Cortex-A55 cores with PSCI enable-method
- GICv3 interrupt controller
- ARMv8 timer
- UART2 and UART3 (disabled by default)

Signed-off-by: Guowei Li <15035660024@163.com>

KouweiLee commented:
Sorry, I have fixed problems again and when I run `./scripts/ci/check_compliance.py -c upstream/main..`, the output is this:
<img width="1624" height="1147" alt="image" src="https://github.com/user-attachments/assets/824c08bc-12e8-41e1-a72f-5f27f388afd2" />
It seems that I don't have to fix these issues?

kartben reviewed (changes requested):
Documentation can be tidied up and, more importantly, would really recommend going for "ROC-RK3588-PC" since it doesn't look like "ROC-RK3588" is an actual product, but I might have missed something. Thanks for the PR!

KouweiLee committed:
board: arm64: Add suport for ROC-RK3588-PC

Add initial support for the ROC-RK3588-PC AArch64 board.
It features a quad-core Cortex-A55 CPU based on the ARMv8.2 architecture.
This commit also enables SMP, allowing all four cores to run the
synchronization sample.

Signed-off-by: Guowei Li <15035660024@163.com>

KouweiLee commented:
> Documentation can be tidied up and, more importantly, would really recommend going for "ROC-RK3588-PC" since it doesn't look like "ROC-RK3588" is an actual product, but I might have missed something. Thanks for the PR!

ok, I have changed ROC-RK3588 to ROC-RK3588-PC in all related files and removed the contents you mentioned in doc. Thanks for advise!

kartben commented:
FWIW you may want to add a board image but this can be done as a follow up

kartben commented on soc/rockchip/rk35/rk3588/mmu_regions.c:
```diff
@@ -0,0 +1,25 @@
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("GIC", DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 0),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
+
+	MMU_REGION_FLAT_ENTRY("GIC", DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 1),
+			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 1),
+			      MT_DEVICE_nGnRnE | MT_P_RW_U_RW | MT_NS),
```
I think it should be `MT_P_RW_U_NA` -- see also https://github.com/zephyrproject-rtos/zephyr/pull/91850 which it would be great to get approved and merged at some point :)

```suggestion
	MMU_REGION_FLAT_ENTRY("GIC", DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 0),
			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 0),
			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),

	MMU_REGION_FLAT_ENTRY("GIC", DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 1),
			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 1),
			      MT_DEVICE_nGnRnE | MT_P_RW_U_NA | MT_NS),
```
